src/python/src/setup.py

```diff
@@ -56,15 +78,47 @@
 if not "darwin" in sys.platform:
     _EXTENSION_LIBRARIES += ('rt',)
 
-_EXTENSION_MODULE = _core.Extension(
-    'grpc._adapter._c', sources=list(_EXTENSION_SOURCES),
+
+_C_EXTENSION_MODULE = _core.Extension(
+    'grpc._adapter._c', sources=list(_C_EXTENSION_SOURCES),
     include_dirs=list(_EXTENSION_INCLUDE_DIRECTORIES),
     libraries=list(_EXTENSION_LIBRARIES),
-    )
+)
+_C_EXTENSION_MODULES = [_C_EXTENSION_MODULE]
+
+
+def cython_extensions(package_names, module_names, include_dirs, libraries,
+                      build_with_cython=False):
+  file_extension = 'pyx' if build_with_cython else 'c'
+  package_files = [name.replace('.', '/') + '/__init__.' + file_extension
+                   for name in package_names]
+  module_files = [name.replace('.', '/') + '.' + file_extension
+                  for name in module_names]
+  extensions = [
+      _extension.Extension(
+          name=module_names[i], sources=[module_files[i]],
+          include_dirs=include_dirs, libraries=libraries
+      ) for i in range(len(module_names))
```

Comment:
"for module_name, module_file in zip(module_names, module_files)" is an option here. Otherwise since you don't use module_files except inside this list comprehension, consider eliminating it and just using an expression in the "sources=" part of this?